### ISSUE TITLE
X Window System error - BadAccess

### ISSUE THREAD

**Issue by ghost** [qTox/qTox#2640]
Got this today, I don't know if it's of any use. Another qTox instance didn't crash.

Debian testing
Linux 4.2.0-1-amd64
Cinnamon 2.6.13
Qt 5.5.1

> [20:41:31.081] src/persistence/settings.cpp:120 : Debug: "Loading settings from /home/user/.config/tox/qtox.ini"
> [20:41:31.099] src/ipc.cpp:46 : Debug: Our global IPC ID is  123
> [20:41:31.099] src/ipc.cpp:64 : Debug: Attaching to the global shared memory
> [20:41:31.099] src/main.cpp:143 : Debug: built on:  Dec  7 2015 ( 1449492671 )
> [20:41:31.099] src/main.cpp:144 : Debug: commit:  89def77ee606e720386ececc455c68c2b106acd1 
> 
> [20:41:31.099] src/main.cpp:171 : Debug: Setting profile to "bob"
> [20:41:31.110] src/nexus.cpp:82 : Debug: Starting up
> [20:41:46.461] src/persistence/profile.cpp:77 : Debug: Loading tox save  "/home/user/.config/tox/bob.tox"
> [20:41:46.610] src/persistence/settings.cpp:346 : Debug: "Saving global settings at /home/user/.config/tox/qtox.ini"
> [20:41:46.650] src/persistence/settings.cpp:287 : Debug: Loading personnal settings from "/home/user/.config/tox/bob.ini"
> [20:41:46.944] src/core/core.cpp:245 : Debug: Loading user profile
> [20:41:46.944] src/persistence/profile.cpp:239 : Debug: Loading tox save  "/home/user/.config/tox/bob.tox"
> [20:41:47.020] src/core/core.cpp:136 : Warning: Core starting with IPv6 disabled. LAN discovery may not work properly.
> [20:41:47.194] src/core/coreencryption.cpp:83 : Warning: Decryption failed
> [20:41:47.194] src/persistence/db/encrypteddb.cpp:104 : Warning: pullFileContent: Encrypted history log is corrupted: can't decrypt, will be deleted
> [20:41:47.194] :0 : Warning: QSqlQuery::value: not positioned on a valid record
> [20:41:47.194] src/persistence/historykeeper.cpp:149 : Debug: Struct DB updated: Added column av_hash in table aliases.
> [20:41:47.194] :0 : Warning: QSqlQuery::value: not positioned on a valid record
> [20:41:47.194] src/persistence/historykeeper.cpp:158 : Debug: Struct DB updated: Added column avatar in table aliases.
> [20:41:47.248] src/core/coreencryption.cpp:83 : Warning: Decryption failed
> [20:41:47.248] src/core/core.cpp:313 : Debug: Self avatar not found, will broadcast empty avatar to friends
> [20:41:47.330] src/core/coreencryption.cpp:83 : Warning: Decryption failed
> [20:41:47.736] src/widget/systemtrayicon.cpp:65 : Debug: Using GTK backend
> 
> (qtox:5610): GLib-GObject-CRITICAL **: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
> [20:41:48.783] src/core/core.cpp:415 : Debug: "Connecting to 212.71.252.109:33445 (Kr9r0x)"
> [20:41:48.783] src/core/core.cpp:415 : Debug: "Connecting to 194.249.212.109:33445 (fluke571)"
> [20:41:53.282] src/core/core.cpp:415 : Debug: "Connecting to 185.25.116.107:33445 (MAH69K)"
> [20:41:53.283] src/core/core.cpp:415 : Debug: "Connecting to 192.99.168.140:33445 (WIeschie)"
> [20:41:56.239] src/core/core.cpp:382 : Debug: Connected to the DHT
> [20:41:56.240] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [20:42:09.009] src/core/coreencryption.cpp:83 : Warning: Decryption failed
> [20:42:09.009] src/core/corefile.cpp:292 : Debug: "Received avatar request 5:65536, accept, since we don't have it in cache."
> [20:42:17.832] src/core/corefile.cpp:439 : Debug: Got 47493 bytes of avatar data from 5
> [20:47:54.327] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [22:45:13.641] src/widget/widget.cpp:1629 : Debug: auto away activated at "22:45:13"
> [22:45:13.641] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [22:45:17.634] src/widget/widget.cpp:1638 : Debug: auto away deactivated at "22:45:17"
> [22:45:17.634] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [23:58:26.635] src/widget/widget.cpp:1629 : Debug: auto away activated at "23:58:26"
> [23:58:26.673] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [23:59:12.633] src/widget/widget.cpp:1638 : Debug: auto away deactivated at "23:59:12"
> [23:59:12.633] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [00:09:37.635] src/widget/widget.cpp:1629 : Debug: auto away activated at "00:09:37"
> [00:09:37.635] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [10:13:21.634] src/widget/widget.cpp:1638 : Debug: auto away deactivated at "10:13:21"
> [10:13:21.635] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [10:46:32.635] src/widget/widget.cpp:1629 : Debug: auto away activated at "10:46:32"
> [10:46:32.635] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [10:47:31.691] src/widget/widget.cpp:1638 : Debug: auto away deactivated at "10:47:31"
> [10:47:31.692] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [11:25:22.635] src/widget/widget.cpp:1629 : Debug: auto away activated at "11:25:22"
> [11:25:22.635] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [13:16:39.666] src/widget/widget.cpp:1638 : Debug: auto away deactivated at "13:16:39"
> [13:16:39.666] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [13:31:02.494] :0 : Warning: QXcbWindow: Unhandled client message: "_GTK_LOAD_ICONTHEMES"
> [13:31:02.495] :0 : Warning: QXcbWindow: Unhandled client message: "_GTK_LOAD_ICONTHEMES"
> [13:31:02.495] :0 : Warning: QXcbWindow: Unhandled client message: "_GTK_LOAD_ICONTHEMES"
> [13:57:28.635] src/widget/widget.cpp:1629 : Debug: auto away activated at "13:57:28"
> [13:57:28.635] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [14:15:36.877] src/widget/widget.cpp:1638 : Debug: auto away deactivated at "14:15:36"
> [14:15:36.877] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [14:15:55.477] :0 : Warning: QXcbWindow: Unhandled client message: "_GTK_LOAD_ICONTHEMES"
> [14:15:55.478] :0 : Warning: QXcbWindow: Unhandled client message: "_GTK_LOAD_ICONTHEMES"
> [14:15:55.478] :0 : Warning: QXcbWindow: Unhandled client message: "_GTK_LOAD_ICONTHEMES"
> [15:04:47.633] src/widget/widget.cpp:1629 : Debug: auto away activated at "15:04:47"
> [15:04:47.633] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [15:25:29.633] src/widget/widget.cpp:1638 : Debug: auto away deactivated at "15:25:29"
> [15:25:29.633] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [17:46:33.633] src/widget/widget.cpp:1629 : Debug: auto away activated at "17:46:33"
> [17:46:33.633] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [17:51:59.876] src/widget/widget.cpp:1638 : Debug: auto away deactivated at "17:51:59"
> [17:51:59.880] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [18:21:13.135] src/widget/widget.cpp:1629 : Debug: auto away activated at "18:21:13"
> [18:21:13.135] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [18:39:56.178] src/widget/widget.cpp:1638 : Debug: auto away deactivated at "18:39:56"
> [18:39:56.178] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [19:01:23.135] src/widget/widget.cpp:1629 : Debug: auto away activated at "19:01:23"
> [19:01:23.135] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [19:26:13.134] src/widget/widget.cpp:1638 : Debug: auto away deactivated at "19:26:13"
> [19:26:13.135] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [20:09:10.135] src/widget/widget.cpp:1629 : Debug: auto away activated at "20:09:10"
> [20:09:10.135] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [21:00:23.135] src/widget/widget.cpp:1638 : Debug: auto away deactivated at "21:00:23"
> [21:00:23.135] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> The program 'qtox' received an X Window System error.
> This probably reflects a bug in the program.
> The error was 'BadAccess (attempt to access private resource denied)'.
>   (Details: serial 4637 error_code 10 request_code 131 minor_code 46)
>   (Note to programmers: normally, X errors are reported asynchronously;
>    that is, you will receive the error a while after causing it.
>    To debug your program, run it with the --sync command line
>    option to change this behavior. You can then get a meaningful
>    backtrace from your debugger if you break on the gdk_x_error() function.)

**Comment by tux3** [comment]
Try updating your qTox version, this crash is probably fixed already

**Comment by ghost** [comment]
Closing because I think it was a one-time thing. Will re-open if it happens with updated commits.

**Comment by ghost** [comment]
Happened again, I'm afraid. Only one instance affected, same one as before. Computer was idle at crash.

> [02:45:30.622] src/persistence/settings.cpp:120 : Debug: "Loading settings from /home/user/.config/tox/qtox.ini"
> [02:45:30.623] src/ipc.cpp:46 : Debug: Our global IPC ID is  123
> [02:45:30.623] src/ipc.cpp:64 : Debug: Attaching to the global shared memory
> [02:45:30.623] src/main.cpp:143 : Debug: built on:  Dec 10 2015 ( 1449711788 )
> [02:45:30.623] src/main.cpp:144 : Debug: commit:  7f4f0bbb82d1c387d042de2957142197cc83cd82 
> 
> [02:45:30.623] src/main.cpp:171 : Debug: Setting profile to "bob"
> [02:45:30.624] src/nexus.cpp:82 : Debug: Starting up
> [02:45:35.044] src/persistence/profile.cpp:77 : Debug: Loading tox save  "/home/user/.config/tox/bob.tox"
> [02:45:35.222] src/persistence/settings.cpp:346 : Debug: "Saving global settings at /home/user/.config/tox/qtox.ini"
> [02:45:35.290] src/persistence/settings.cpp:287 : Debug: Loading personnal settings from "/home/user/.config/tox/bob.ini"
> [02:45:35.545] src/core/core.cpp:245 : Debug: Loading user profile
> [02:45:35.545] src/persistence/profile.cpp:239 : Debug: Loading tox save  "/home/user/.config/tox/bob.tox"
> [02:45:35.630] src/core/core.cpp:136 : Warning: Core starting with IPv6 disabled. LAN discovery may not work properly.
> [02:45:35.737] src/persistence/db/encrypteddb.cpp:43 : Debug: writing old data
> [02:45:35.737] :0 : Warning: QSqlQuery::value: not positioned on a valid record
> [02:45:35.738] src/persistence/historykeeper.cpp:149 : Debug: Struct DB updated: Added column av_hash in table aliases.
> [02:45:35.738] :0 : Warning: QSqlQuery::value: not positioned on a valid record
> [02:45:35.738] src/persistence/historykeeper.cpp:158 : Debug: Struct DB updated: Added column avatar in table aliases.
> [02:45:36.461] src/widget/systemtrayicon.cpp:65 : Debug: Using GTK backend
> 
> (qtox:5195): GLib-GObject-CRITICAL **: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
> [02:45:37.335] src/core/core.cpp:415 : Debug: "Connecting to 185.25.116.107:33445 (MAH69K)"
> [02:45:37.336] src/core/core.cpp:415 : Debug: "Connecting to 192.99.168.140:33445 (WIeschie)"
> [02:45:40.353] src/core/core.cpp:382 : Debug: Connected to the DHT
> [02:45:40.353] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [02:45:59.433] src/core/corefile.cpp:270 : Debug: "Received empty avatar request 5:65536"
> [02:45:59.683] src/core/corefile.cpp:337 : Debug: Avatar transfer 0 to friend 5 accepted
> [02:46:00.591] src/core/corefile.cpp:270 : Debug: "Received empty avatar request 5:131072"
> [02:46:00.795] src/core/corefile.cpp:337 : Debug: Avatar transfer 1 to friend 5 accepted
> [02:46:17.601] src/core/corefile.cpp:270 : Debug: "Received empty avatar request 5:196608"
> [02:46:17.794] src/core/corefile.cpp:270 : Debug: "Received empty avatar request 5:262144"
> [02:46:36.632] src/core/corefile.cpp:270 : Debug: "Received empty avatar request 5:327680"
> [03:11:35.634] src/widget/widget.cpp:1629 : Debug: auto away activated at "03:11:35"
> [03:11:35.634] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [03:11:35.947] src/core/corefile.cpp:270 : Debug: "Received empty avatar request 5:393216"
> [09:18:13.635] src/widget/widget.cpp:1638 : Debug: auto away deactivated at "09:18:13"
> [09:18:13.635] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [09:18:13.873] src/core/corefile.cpp:270 : Debug: "Received empty avatar request 5:458752"
> [09:30:39.635] src/widget/widget.cpp:1629 : Debug: auto away activated at "09:30:39"
> [09:30:39.635] src/persistence/profile.cpp:304 : Debug: Saving tox save to  "/home/user/.config/tox/bob.tox"
> [09:30:40.049] src/core/corefile.cpp:270 : Debug: "Received empty avatar request 5:524288"
> The program 'qtox' received an X Window System error.
> This probably reflects a bug in the program.
> The error was 'BadAccess (attempt to access private resource denied)'.
>   (Details: serial 927 error_code 10 request_code 131 minor_code 46)
>   (Note to programmers: normally, X errors are reported asynchronously;
>    that is, you will receive the error a while after causing it.
>    To debug your program, run it with the --sync command line
>    option to change this behavior. You can then get a meaningful
>    backtrace from your debugger if you break on the gdk_x_error() function.)

**Comment by sudden6** [comment]
maybe related to #2615

**Comment by agilob** [comment]
#1277  #1564

```
The program 'qtox' received an X Window System error.
This probably reflects a bug in the program.
The error was 'BadAccess (attempt to access private resource denied)'.
  (Details: serial 641 error_code 10 request_code 131 minor_code 46)
  (Note to programmers: normally, X errors are reported asynchronously;
   that is, you will receive the error a while after causing it.
   To debug your program, run it with the --sync command line
   option to change this behavior. You can then get a meaningful
   backtrace from your debugger if you break on the gdk_x_error() function.)
```

**Comment by tux3** [comment]
@agilob that log you posted shows qTox exiting normally because it doesn't know the "--sync" option, that's not a crash.

**Comment by ghost** [comment]
I think this might be related to #2144. After hot swapping a few times the external monitor I got the

> The program 'qtox' received an X Window System error.
> This probably reflects a bug in the program.
> The error was 'BadAccess (attempt to access private resource denied)'.
>   (Details: serial 995 error_code 10 request_code 131 minor_code 46)
>   (Note to programmers: normally, X errors are reported asynchronously;
>    that is, you will receive the error a while after causing it.
>    To debug your program, run it with the --sync command line
>    option to change this behavior. You can then get a meaningful
>    backtrace from your debugger if you break on the gdk_x_error() function.)

message. The fact that It crashed overnight may be the monitor going on energy save modus. Interesting facts: 
- it took a couple of tries to make qTox crash
- still only one instance crashes (this time a different profile)
- further hot swapping didn't make the second instance crash

Running 18484cb9502ebc8bf4dc161fb243f2dc657ac7b7, self compiled, Qt 5.5.1. On #2144 I was using the pre-compiled binaries and Qt 5.2.1.

**Comment by agilob** [comment]
That's a lot of useful info, I'll take a look at this bug when my screen arrives!

**Comment by BOPOHA** [comment]
I'm affected.
It happens when i insert the USB wireless stick (for Plantronics Audio 995).

```
[user@host ~]$ qtox -v

(qtox:21176): GLib-GObject-WARNING **: g_object_get_valist: object class 'GtkSettings' has no property named 'gtk-application-prefer-dark-theme'

(qtox:21176): GLib-GObject-WARNING **: g_object_get_valist: object class 'GtkSettings' has no property named 'gtk-long-press-time'
qTox 
Git commit: e6e666fa8ce36cd3792d4536a59ab2b65fd5b546
[user@host ~]$ qtox 

(qtox:20429): GLib-GObject-WARNING **: g_object_get_valist: object class 'GtkSettings' has no property named 'gtk-application-prefer-dark-theme'

(qtox:20429): GLib-GObject-WARNING **: g_object_get_valist: object class 'GtkSettings' has no property named 'gtk-long-press-time'

(qtox:20429): Gdk-ERROR **: The program 'qtox' received an X Window System error.
This probably reflects a bug in the program.
The error was 'BadAccess (attempt to access private resource denied)'.
  (Details: serial 854 error_code 10 request_code 131 minor_code 46)
  (Note to programmers: normally, X errors are reported asynchronously;
   that is, you will receive the error a while after causing it.
   To debug your program, run it with the --sync command line
   option to change this behavior. You can then get a meaningful
   backtrace from your debugger if you break on the gdk_x_error() function.)
Trace/breakpoint trap (core dumped)
[user@host ~]$ 
```

dmesg

> [41395.719195] usb 2-2: new full-speed USB device number 7 using xhci_hcd
> [41395.916030] usb 2-2: New USB device found, idVendor=047f, idProduct=d957
> [41395.916039] usb 2-2: New USB device strings: Mfr=3, Product=4, SerialNumber=0
> [41395.916043] usb 2-2: Product: Plantronics Wireless Audio
> [41395.916046] usb 2-2: Manufacturer: Plantronics Wireless Audio
> [41396.060253] input: Plantronics Wireless Audio Plantronics Wireless Audio as /devices/pci0000:00/0000:00:14.0/usb2/2-2/2-2:1.3/0003:047F:D957.0004/input/input31
> [41396.112266] plantronics 0003:047F:D957.0004: input,hiddev0,hidraw1: USB HID v1.11 Device [Plantronics Wireless Audio Plantronics Wireless Audio] on usb-0000:00:14.0-2/input3
> [41396.185638] traps: qtox[21194] trap int3 ip:7ffaf40b819b sp:7ffc9d076500 error:0 in libglib-2.0.so.0.4800.1[7ffaf4068000+10d000]

**Comment by sudden6** [comment]
Assuming fixed in a recent qTox version, if not please open a new issue.

**Comment by anthonybilinski** [comment]
Just a note that this is reproducible on our latest released AppImage, v1.16.3, which was built with Ubuntu 14.04 dependencies. On nightly AppImages, it's no longer reproducible (now built on Ubuntu 16.04).
